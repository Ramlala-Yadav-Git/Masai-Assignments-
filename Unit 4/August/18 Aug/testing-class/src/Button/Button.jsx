
import styles from "./Button.module.css"

export function Button({ label, onClick }) {


    return (<>

        <div data-testid="button" onClick={onClick} className={styles.Button}>
            {label}
        </div>
    </>)
}