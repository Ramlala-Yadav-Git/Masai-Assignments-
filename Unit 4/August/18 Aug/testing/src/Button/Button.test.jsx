
import { fireEvent, render, screen } from "@testing-library/react"
import { Button } from "./Button"
import renderer from "react-test-renderer"

describe("Button.jsx testing", () => {

    test("render learn react link", () => {

        const text = "CLICK ME"
        render(<>
            <Button label={text} />

        </>)
        const btn = screen.getByText(text)
        expect(btn).toBeInTheDocument()

    })
    test("button should render with correct label", () => {

        const text = "CLICK ME"
        render(<>
            <Button label={text} />

        </>)
        const btn = screen.getByTestId('button')
        expect(btn.textContent).toBe(text)

    })

    it('should invoke a function when you click on it', () => {
        const fn = jest.fn()
        const text = "CLICK ME"
        render(<>
            <Button onClick={fn} label={text}></Button>

        </>)

        const btn = screen.getByTestId("button")
        fireEvent.click(btn)

        expect(fn).toHaveBeenCalled()
        expect(fn).toHaveBeenCalledTimes(1)

    })
    it('should match the snapshot', () => {
        const text = "Button"
        const tree = renderer.create(
            <Button onClick={() => { }} label={text} />
        )
        expect(tree).toMatchSnapshot()
    })
})