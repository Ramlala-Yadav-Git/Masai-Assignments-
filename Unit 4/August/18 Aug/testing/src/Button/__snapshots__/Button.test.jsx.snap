// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.jsx testing should match the snapshot 1`] = `
<div
  className="Button"
  data-testId="button"
  onClick={[Function]}
>
  Button
</div>
`;
